lib/protocol/src/openctx-protocol.schema.ts

```diff
@@ -62,11 +79,46 @@ export interface Selector {
      */
     contentContains?: string
 }
+/**
+ * Defines a kind for an item the provider can return.
+ */
+export interface ItemKind {
+    /**
+     * A unique identifier for the kind.
+     */
+    id: string
+    /**
+     * A title for the kind.
+     */
+    title: string
+    /**
+     * An optional description for the kind.
+     */
+    description?: string
+}
+export interface CandidateItemsParams {
+    /**
+     * A search query that is interpreted by providers to filter the items in the result set.
+     */
+    query?: string
+    /**
+     * The kind of item interpreted by providers to return items of the specified kind.
+     */
+    kind?: string
+}
 export interface ItemsParams {
     /**
      * A search query that is interpreted by providers to filter the items in the result set.
      */
     query?: string
+    /**
+     * The kind of item interpreted by providers to return items of the specified kind.
+     */
+    kind?: string
+    /**
+     * The list of candidate items interpreted by providers to return items for the specified candidates.
+     */
+    candidateItems?: Item[]
```

Comment:
I propose this just be a single item, rather than a list. Makes provider implementation more straightforward.

Reply:
Done